File: expensereport-cxx/Expense.cpp
#include "Expense.h"

Expense::Expense(const ExpenseType& type, const int amount) : type(type), amount(amount) {
}

bool Expense::isOverLimit() const {
    return amount > type.limit;
}

bool Expense::isMeal() const {
    return type.isMeal;
}

std::string Expense::name() const {
    return type.name;
}

File: expensereport-java/src/main/java/com/nelkinda/training/Expense.java
package com.nelkinda.training;

public record Expense(ExpenseType type, int amount) {
    public boolean isMeal() {
        return type.isMeal();
    }

    public boolean exceedsLimit() {
        return type.exceedsLimit(amount);
    }

    public String getName() {
        return type.getName();
    }
}
